core/blockchain.go

```diff
@@ -1949,3 +1994,12 @@ func (bc *BlockChain) ResetToStateSyncedBlock(block *types.Block) error {
 	bc.initSnapshot(head)
 	return nil
 }
+
+// FlattenLogs converts a nested array of logs to a single array of logs.
+func FlattenLogs(list [][]*types.Log) []*types.Log {
```

Comment:
nit: could this be unexported?

Reply:
We use this in `eth/filter`. I moved it to avoid duplicating the code.

Reply:
Could we move this to next to `types.Log` ?